test/cluster/config_admin_test.go

```diff
@@ -0,0 +1,296 @@
+	require.Equal(t, alice[0], admins[1])
+	require.Equal(t, bob[0], admins[2])
+
+	bobCertPath, bobKeyPath := c.GetUserCertKeyPath("bob")
+	c.UpdateServersAdmin("bob", bobKeyPath, bobCertPath)
+
+	//bob tries to add Alice again
+	newConfig = configEnv.GetResponse().GetConfig()
+	newConfig.Admins = append(newConfig.Admins, alice...)
+	version = configEnv.GetResponse().GetMetadata().GetVersion()
+	txID, rcpt, err = leaderServer.SetConfigTx(t, newConfig, version, c.Servers[leaderIndex].AdminSigner(), "bob")
+	require.EqualError(t, err, "failed to submit transaction, server returned: status: 400 Bad Request, message: Invalid config tx, reason: there are two admins with the same ID [alice] in the admin config. The admin IDs must be unique")
+}
+
+func TestAddNewAdminWithInvalidCertificate(t *testing.T) {
+	dir, err := ioutil.TempDir("", "int-test")
+	require.NoError(t, err)
+
+	nPort, pPort := getPorts(1)
+	setupConfig := &setup.Config{
+		NumberOfServers:     1,
+		TestDirAbsolutePath: dir,
+		BDBBinaryPath:       "../../bin/bdb",
+		CmdTimeout:          10 * time.Second,
+		BaseNodePort:        nPort,
+		BasePeerPort:        pPort,
+		CheckRedirectFunc: func(req *http.Request, via []*http.Request) error {
+			return errors.Errorf("Redirect blocked in test client: url: '%s', referrer: '%s', #via: %d", req.URL, req.Referer(), len(via))
+		},
+	}
+	c, err := setup.NewCluster(setupConfig)
+	require.NoError(t, err)
+	defer c.ShutdownAndCleanup()
+
+	require.NoError(t, c.Start())
+
+	leaderIndex := -1
+	require.Eventually(t, func() bool {
+		leaderIndex = c.AgreedLeader(t, 0)
+		return leaderIndex >= 0
+	}, 30*time.Second, 100*time.Millisecond)
+	leaderServer := c.Servers[leaderIndex]
+
+	alice := []*types.Admin{
+		{
+			Id:          "alice",
+			Certificate: []byte("invalid certificate"),
+		},
+	}
+
+	//get current cluster config
+	configEnv, err := leaderServer.QueryConfig(t)
+	require.NoError(t, err)
+	require.NotNil(t, configEnv)
+	newConfig := configEnv.GetResponse().GetConfig()
+
+	newConfig.Admins = append(newConfig.Admins, alice...)
+	version := configEnv.GetResponse().GetMetadata().GetVersion()
+
+	//add alice with invalid certificate
+	_, _, err = leaderServer.SetConfigTx(t, newConfig, version, c.Servers[leaderIndex].AdminSigner(), "admin")
+	require.EqualError(t, err, "failed to submit transaction, server returned: status: 400 Bad Request, message: Invalid config tx, reason: the admin [alice] has an invalid certificate: error parsing certificate: asn1: structure error: tags don't match (16 vs {class:1 tag:9 length:110 isCompound:true}) {optional:false explicit:false application:false private:false defaultValue:<nil> tag:<nil> stringType:0 timeType:0 set:false omitEmpty:false} certificate @2")
+}
+
+// Scenario:
+// - trying to delete the only admin in the cluster
+// - config tx fails because there must be at least a single admin in the cluster
+// - admin adds alice to admins list
+// - alice deletes admin from the admins' list
```

Comment:
Please add another test case: and admin changes its certificate; i.e.  
1. start normally with `admin` as the admin, 
2. load the cert and signer for `alice`,
3. have admin update itself to have the  cert of `alice` instead of its original cert, 
4. and then try to get the config: 
  - first with admin's old signer (will fail) 
  - with the new signer based on alice's private key (will succeed).

Reply:
Done